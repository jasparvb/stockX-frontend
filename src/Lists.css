.Lists .list-container {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}

.Lists .row {
  background: whitesmoke;
}