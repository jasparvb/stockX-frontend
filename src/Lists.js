import React, { useState, useEffect } from "react";
import './Lists.css';
import { useHistory } from "react-router-dom";
import { useDispatch, useSelector } from 'react-redux';
import { addAlert } from './actions/alerts';
import { getListsAPI } from './actions/lists';
import { removeListAPI, removeStockAPI } from './actions/lists';
import AddListForm from './AddListForm';
import List from './List';

function Lists() {
  const dispatch = useDispatch();
  const history = useHistory();
  const token = useSelector(st => st.users);
  const lists = useSelector(st => st.lists);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    async function getLists() {
      await dispatch(getListsAPI());
      setIsLoading(false);
    }

    if (isLoading) {
      getLists();
    }
    
  }, [dispatch, isLoading, lists]);

  async function removeList(id) {
    await dispatch(removeListAPI(id));
    setIsLoading(true);
  }

  async function removeStock(id, listId) {
    await dispatch(removeStockAPI(id, listId));
    setIsLoading(true);
  }

  if (isLoading) return <h3><b>Loading...</b></h3>;
  
  if (!isLoading && !lists) return <h1>You don't have any posts yet</h1>;


  //bounces user to home page if not logged in
  if (!token) {
    dispatch(addAlert(`You must be logged in to view that page!`, "danger"));
    history.push('/login');
  }
  return (
    <div className="Lists container-fluid">
      <div className="container text-left col-md-6 offset-md-3 col-lg-4 offset-lg-4 my-3">
        <h3>Enter a title to create a new list</h3>
        <p>Once you create a list, you can search for stocks and add them to your list for easy tracking.</p>
        <AddListForm />
      </div>
      {lists ? 
      <div className="list-container">
        {lists.map(l => (
          <List 
            key={l.id} 
            id={l.id} 
            name={l.name} 
            stocks={l.stocks} 
            removeList={removeList} 
            removeStock={removeStock}
          />
        ))}
      </div>
      : <div className="text-center my-5">
        <h4>(You haven't created any lists yet)</h4>
      </div>}
    </div>
  );
}

export default Lists;