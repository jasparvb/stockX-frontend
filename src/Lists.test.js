import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from "react-redux";
import rootReducer from "./reducers/root";
import { BrowserRouter } from "react-router-dom";
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from 'redux-devtools-extension';
import Lists from './Lists';

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)
  )
);

it("renders without crashing", function() {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Lists />
      </BrowserRouter>
    </Provider>);
});

it("matches snapshot", function() {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Lists />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
