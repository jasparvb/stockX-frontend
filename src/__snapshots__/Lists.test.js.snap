// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h3>
    <b>
      Loading...
    </b>
  </h3>
</DocumentFragment>
`;
